typing_extensions/src/typing_extensions.py

```diff
@@ -91,6 +73,88 @@ def _check_generic(cls, parameters):
 if PEP_560:
     __all__.extend(["get_args", "get_origin", "get_type_hints"])
 
+# The functions below are modified copies of typing internal helpers.
+# They are needed by _ProtocolMeta and they provide support for PEP 646.
+
+
+def _no_slots_copy(dct):
+    dict_copy = dict(dct)
+    if '__slots__' in dict_copy:
+        for slot in dict_copy['__slots__']:
+            dict_copy.pop(slot, None)
+    return dict_copy
+
+
+_marker = object()
+
+
+def _check_generic(cls, parameters, elen=_marker):
```

Comment:
I think Mathew pointed out in his documenting typing.py PR that `parameters` is confusing. Should we use `args` or `type_args` instead?

Also what's `elen`'s default for? When would we not want to pass in `__parameters__` ourselves?

Reply:
This is a modified copy of typing.py's `_check_generic`, and it's monkeypatched into typing later. We probably shouldn't modify the function's interface, because that makes the monkeypatching even riskier.